internal/telemetry/client.go

```diff
@@ -229,6 +233,16 @@ func (c *client) start(configuration []Configuration, namespace Namespace) {
 	}
 	c.heartbeatInterval = time.Duration(heartbeat) * time.Second
 	c.heartbeatT = time.AfterFunc(c.heartbeatInterval, c.backgroundHeartbeat)
+
+	// app closing should be sent when the app closes gracefully
+	c.sigc = make(chan os.Signal, 1)
+	signal.Notify(c.sigc, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
```

Comment:
Couldn't this interfere with other signal handlers that the user has set up for their program?

Reply:
Ah, nevermind, I should read the docs 😆 https://pkg.go.dev/os/signal#Notify

> It is allowed to call Notify multiple times with different channels and the same signals: each channel receives copies of incoming signals independently.